tests/ts/query-builder-api/mutating-methods.ts

```diff
@@ -0,0 +1,207 @@
+import { raw, ref } from '../../../';
+import { Movie } from '../fixtures/movie';
+import { Person } from '../fixtures/person';
+
+(async () => {
+  await Person.query().insert({ firstName: 'Jennifer', lastName: 'Lawrence' });
+
+  await Movie.relatedQuery('actors')
+    .for(1)
+    .insert([
+      { firstName: 'Jennifer', lastName: 'Lawrence' },
+      { firstName: 'Bradley', lastName: 'Cooper' }
+    ]);
+
+  await Person.query().insert({
+    age: Person.query().avg('age'),
+    firstName: raw("'Jenni' || 'fer'")
+  });
+
+  await Person.query()
+    .insert({
+      age: Person.query().avg('age'),
+      firstName: raw("'Jenni' || 'fer'")
+    })
+    .returning('*');
+
+  await Movie.relatedQuery('actors')
+    .for(1)
+    .insert({
+      firstName: 'Jennifer',
+      lastName: 'Lawrence',
+      someExtra: "I'll be written to the join table"
+    });
+
+  await Person.query().insertAndFetch({ firstName: 'Jennifer', lastName: 'Lawrence' });
+
+  await Movie.relatedQuery('actors')
+    .for(1)
+    .insertAndFetch([
+      { firstName: 'Jennifer', lastName: 'Lawrence' },
+      { firstName: 'Bradley', lastName: 'Cooper' }
+    ]);
+
+  await Person.query().insertAndFetch({
+    age: Person.query().avg('age'),
+    firstName: raw("'Jenni' || 'fer'")
+  });
+
+  await Movie.relatedQuery('actors')
+    .for(1)
+    .insertAndFetch({
+      firstName: 'Jennifer',
+      lastName: 'Lawrence',
+      someExtra: "I'll be written to the join table"
+    });
+
+  await Person.query().insertGraphAndFetch({ firstName: 'Jennifer', lastName: 'Lawrence' });
+
+  await Person.query()
+    .patch({ age: 24 })
+    .findById(1);
+
+  await Person.query()
+    .patch({ age: 20 })
+    .where('age', '<', 50);
+
+  await Person.query()
+    .patch({ age: raw('age + 1') })
+    .where('age', '<', 50);
+
+  await Person.query().patch({
+    age: Person.query().avg('age'),
+    // You can use knex.raw instead of `raw()` if
+    // you prefer.
+    firstName: raw("'Jenni' || 'fer'"),
+    oldLastName: ref('lastName'),
+    // This updates a value nested deep inside a
+    // json column `detailsJsonColumn`.
+    'detailsJsonColumn:address.street': 'Elm street'
```

Comment:
The typings currently don't support this syntax, any thoughts? I am not sure if there is a way in TypeScript to type string object access like this....

Reply:
Yeah, I don't think we csn make the typings support this syntax. We could allow any property there, but it would weaken the typings for most people. We probably should add another syntax for updating json columns at some point. For now, you can just leave those out.